### PR TITLE
pmyip3/Remove user schema completely 

### PR DESCRIPTION
**Pull request by pmyip3**
# Description 
- remove user schema and modify relevant model depends on it (both backend and frontend files)
- remove the teamId from the route for publication page
- update the team schema

# Type of change 
- [ ] Bug fix
- [ ] New feature
- [X] Refactoring
- [ ] Documentation 

# Problem
The team has decided to remove the user schema as all the team members from a research team will be using the same account to login. 

# Solution description
Remove the user schema and let the team schema responsible for storing the team info. 

# Tests 
All unit tests pass? N/A
Builds successfully? Yes 

Attach screenshot of the expected result (e.g. UI of the website/ postman api call result/ database changes)
Changed route path for publication (removed teamId)
image

Provide steps in detail for reviewers to reproduce the expected result of the code changes.

# Relevant document/ link (if any)
link for updated team model  
https://www.notion.so/Researchify-DB-Schema-78d164ce0c6147f9831f79cd9b89c519

# Risk
Identity the risk level (low, medium, high) and indicate which component(s) will be affected if bugs exist.
Low

### CODE REVIEW

**Review by cyr17 — APPROVED** [review]
looks good!

File: api/src/models/team.model.js [hunk 1]
```diff
@@ -22,14 +22,6 @@ const teamSchema = new mongoose.Schema(
       type: String,
       required: true,
     },
-    areaOfInterest: {
```
Comment (riordanalfredo) [inline api/src/models/team.model.js:25]: I might have missed something... why do we want to remove this?

File: client/src/components/App.js [hunk 2]
```diff
@@ -15,24 +15,16 @@ import Login from './auth/Login';
...
 const App = () => {
   const urls = {
     dashboard: '/dashboard',
     profile: '/dashboard/profile',
   };
-  const teamId = useSelector((state) => state.team.teamId);
-  const dispatch = useDispatch();
-
-  useEffect(() => {
-    dispatch(fetchUserAction());
```
Comment (riordanalfredo) [inline client/src/components/App.js:34]: Unfortunately, we need this one. My mistake! I put a confusing action name (it was before our full discussion about remove users totally) Let me revert it back and give an appropriate name ;)

**Review by riordanalfredo — COMMENTED** [review]
Great work! There are some changes that need to be made (I'll help out)

File: client/src/components/App.js (hunk 2 above)
Comment (riordanalfredo) [reply client/src/components/App.js:34]: Resolved!

**Review by riordanalfredo — APPROVED** [review]
It looks good to me :)

File: api/src/models/team.model.js (hunk 1 above)
Comment (pmyip3) [reply api/src/models/team.model.js:25]: I think I also missed the discussion. I just refer to the model in this link: https://www.notion.so/Researchify-DB-Schema-78d164ce0c6147f9831f79cd9b89c519

**Review by vgoh1 — APPROVED** [review]

**Pull request merged**